test/last-run.js

```diff
@@ -45,4 +45,99 @@ describe('lastRun', function(){
+    taker.emit('error', {uid: 1, name: 'test1'});
+
+    expect(taker.lastRun('test1')).to.not.exist();
+    done();
+  });
+
+  it('should keep last record task start time on error', function(done){
+    var since = Date.now();
+
+    taker.emit('start', {uid: 1, name: 'test1', time: since});
+    taker.emit('stop', {uid: 1, name: 'test1', time: since + 1});
+    taker.emit('start', {uid: 1, name: 'test1', time: since + 2});
+    taker.emit('error', {uid: 1, name: 'test1'});
+
+    expect(taker.lastRun('test1')).to.be.equal(since);
+    done();
+  });
+
+  it('should record last task start time on concurrent run', function(done){
+    var since = Date.now();
+
+    taker.emit('start', {uid: 1, name: 'test1', time: since - 1});
+    taker.emit('start', {uid: 2, name: 'test1', time: since});
+    taker.emit('stop', {uid: 2, name: 'test1', time: since + 1});
+    taker.emit('stop', {uid: 1, name: 'test1', time: since + 2});
+
+    expect(taker.lastRun('test1')).to.be.equal(since);
+    done();
+  });
+
+  it('should record last task start time on concurrent run with error', function(done){
+    var since = Date.now();
+
+    taker.emit('start', {uid: 1, name: 'test1', time: since - 1});
+    taker.emit('start', {uid: 2, name: 'test1', time: since});
+    taker.emit('stop', {uid: 2, name: 'test1', time: since + 1});
+    taker.emit('error', {uid: 1, name: 'test1'});
+
+    expect(taker.lastRun('test1')).to.be.equal(since);
+    done();
+  });
+
+  it('should record last task start time on concurrent run with error first', function(done){
+    var since = Date.now();
+
+    taker.emit('start', {uid: 1, name: 'test1', time: since - 1});
+    taker.emit('start', {uid: 2, name: 'test1', time: since});
+    taker.emit('error', {uid: 1, name: 'test1'});
+    taker.emit('stop', {uid: 2, name: 'test1', time: since + 1});
+
+    expect(taker.lastRun('test1')).to.be.equal(since);
+    done();
+  });
+
+  it('keep last record task start time on unexpected error', function(done){
+    var since = Date.now();
+
+    // stop event without any start event
+    taker.emit('stop', {uid: 1, name: 'test1', time: since - 4});
+    expect(taker.lastRun('test1')).to.not.exist();
+
+    // error event without any start event
+    taker.emit('error', {uid: 2, name: 'test1', time: since - 3});
+    expect(taker.lastRun('test1')).to.not.exist();
+
+    // Start event without start time
+    taker.emit('start', {uid: 3, name: 'test1'});
+    taker.emit('stop', {uid: 3, name: 'test1', time: since - 1});
+    expect(taker.lastRun('test1')).to.not.exist();
+
+    taker.emit('start', {uid: 4, name: 'test1', time: since});
+    taker.emit('stop', {uid: 4, name: 'test1', time: since + 1});
+    expect(taker.lastRun('test1')).to.be.equal(since);
+
+    // stop event without matching start event
+    taker.emit('stop', {uid: 5, name: 'test1', time: since + 2});
+    expect(taker.lastRun('test1')).to.be.equal(since);
+
+    done();
+  });
```

Comment:
That last test case in only meant to get 100% coverage (there must be a start event matching a stop or error event).

Maybe I should not test if `_lastRuns[task]` is set in the error and stop handler instead.